packages/smart-contracts/test/contracts/EthereumFeeProxy.test.ts

```diff
@@ -45,6 +45,27 @@ describe('contract: EthereumFeeProxy', () => {
     expect(contractBalance.toString()).to.equals("0");
   });
 
+  it('allows to pays exact eth with a reference with extra msg.value', async () => {
```

Comment:
It would be nice to test the behaviour when:
- `msg.value` < `amount`
- `msg.value` > `amount`